compiler/frontend/src/mir_optimize/reference_following.rs

```diff
@@ -28,15 +28,8 @@ pub fn follow_references(context: &mut Context, expression: &mut CurrentExpressi
 }
 
 pub fn remove_redundant_return_references(body: &mut Body) {
-    while body.expressions.len() > 1 {
-        let [.., (second_last_id, _), (_, last_expression)] = &body.expressions[..] else {
-            unreachable!()
-        };
-
-        if let Expression::Reference(referenced) = last_expression && referenced == second_last_id {
-            body.expressions.pop();
-        } else {
-            break;
-        }
+    while let [.., (second_last_id, _), (_, Expression::Reference(referenced))] = &body.expressions[..]
+        && referenced == second_last_id {
```

Comment:
Would this also work?

```suggestion
    while let [.., (id, _), (_, Expression::Reference(id))] = &body.expressions[..] {
```

Reply:
I tried this, but Rust complained about two captures with the same name